From 7a7c5469ef4ce7023950255a2ce44b9dcef59588 Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Sat, 11 Dec 2021 15:13:34 -0500
Subject: [PATCH] [HACK] cmd: ums: Ensure USB gadget is probed via workaround

The actual solution would be to ensure probing happens in
`g_dnl_register`, but forcing a call to
`g_dnl_board_usb_cable_connected` is inexpensive and *anyway* something
that happens later.
---
 cmd/usb_mass_storage.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/cmd/usb_mass_storage.c b/cmd/usb_mass_storage.c
index 14fa7233c7..f4e891e9e3 100644
--- a/cmd/usb_mass_storage.c
+++ b/cmd/usb_mass_storage.c
@@ -177,6 +177,13 @@ static int do_usb_mass_storage(struct cmd_tbl *cmdtp, int flag,
 		goto cleanup_board;
 	}
 
+#if defined(CONFIG_USB_MUSB_GADGET) && !CONFIG_IS_ENABLED(DM_USB_GADGET)
+	// Hack for sunxi musb not being probed eagerly.
+	// Using `g_dnl_board_usb_cable_connected` ends up probing the cable.
+	// This is a workaround, and `g_dnl_register` should instead probe correctly.
+	g_dnl_board_usb_cable_connected();
+#endif
+
 	rc = g_dnl_register("usb_dnl_ums");
 	if (rc) {
 		pr_err("g_dnl_register failed\n");
-- 
2.34.0

